hbase/src/main/java/org/apache/metamodel/hbase/HBaseUpdateCallback.java

```diff
@@ -0,0 +1,112 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.hbase;
+
+import java.util.List;
+import java.util.Set;
+
+import org.apache.metamodel.AbstractUpdateCallback;
+import org.apache.metamodel.UpdateCallback;
+import org.apache.metamodel.create.TableCreationBuilder;
+import org.apache.metamodel.delete.RowDeletionBuilder;
+import org.apache.metamodel.drop.TableDropBuilder;
+import org.apache.metamodel.insert.RowInsertionBuilder;
+import org.apache.metamodel.schema.Schema;
+import org.apache.metamodel.schema.Table;
+
+/**
+ * This class is used to build objects to do client-operations on a HBase datastore
+ */
+public class HBaseUpdateCallback extends AbstractUpdateCallback implements UpdateCallback {
+
+    public HBaseUpdateCallback(final HBaseDataContext dataContext) {
+        super(dataContext);
+    }
+
+    @Override
+    public TableCreationBuilder createTable(final Schema schema, final String name) {
+        return new HBaseCreateTableBuilder(this, schema, name);
+    }
+
+    /**
+     * Initiates the building of a table creation operation.
+     * @param schema the schema to create the table in
+     * @param name the name of the new table
+     * @param columnFamilies the columnFamilies of the new table
+     * @return {@link HBaseCreateTableBuilder}
+     */
+    public HBaseCreateTableBuilder createTable(final Schema schema, final String name,
+            final Set<String> columnFamilies) {
+        return new HBaseCreateTableBuilder(this, schema, name, columnFamilies);
+    }
+
+    @Override
+    public boolean isDropTableSupported() {
+        return true;
+    }
+
+    @Override
+    public TableDropBuilder dropTable(final Table table) {
+        return new HBaseTableDropBuilder(table, this);
+    }
+
+    /**
+     * @throws UnsupportedOperationException use {@link HBaseUpdateCallback#insertInto(String, String)}
+     */
+    @Override
+    public RowInsertionBuilder insertInto(final Table table) {
+        throw new UnsupportedOperationException(
+                "We need an explicit list of columns when inserting into an HBase table.");
+    }
+
+    /**
+     * Initiates the building of a row insertion operation.
+     * @param table Table to get inserts.
+     * @param columns List of {@link HBaseColumn} to insert on.
+     * @return {@link HBaseRowInsertionBuilder}
+     * @throws IllegalArgumentException The table must be an {@link HBaseTable} and the columns list can't be null or empty
+     */
+    public HBaseRowInsertionBuilder insertInto(final Table table, final List<HBaseColumn> columns) {
```

Comment:
I don't think it makes much sense to expose this method which is not part of the interface, but to make the `insertInto(Table)` method throw `UnsupportedOperationException`. If the user cannot use the interface, then why even implement it? The user will have to do something quite awkward like:

```
dataContext.executeUpdate(cb -> {
  HBaseUpdateCallback hbc = (HBaseUpdateCallback) cb;
  // something goes here
});
```
Normally we even keep the UpdateCallback implementations non-public because we don't want the users to know about it, just the interface.

Reply:
This part has been refactored. You can now use the insertInto(Table) method as exposed by the api and add the columns and values to the returned RowInsertionBuilder.